lcls-twincat-common-components/Library/Devices/PPM/FB_PPM_PowerMeter.TcPOU

```diff
@@ -103,6 +172,26 @@ fbGetPMVoltage(
 fCalibBase := (fVoltage + fCalibRelOffset) * fCalibRelRatio;
 fCalibMJ := fCalibBase * fCalibMJRatio;
 
+// (Re)starting backround collection, will gracefully restart
+IF fBackgroundCollect = TRUE THEN
+    fBackgroundCollect := FALSE;
+    fBackgroundCollectionCycles := 10_000_000 * fBackgroundCollectionTime / _TaskInfo[GETCURTASKINDEXEX()].CycleTime;
+    fBackgroundCollectionCyclesDone := 0;
+    fBackgroundCollectionSum := 0;
+END_IF
+// Keeps a running average of background voltage
+IF fBackgroundCollectionCycles <> 0 THEN
+    fBackgroundCollectionSum := fBackgroundCollectionSum + fVoltage;
+    fBackgroundCollectionCycles := fBackgroundCollectionCycles - 1;
+    fBackgroundCollectionCyclesDone := fBackgroundCollectionCyclesDone + 1;
+    fBackground := fBackgroundCollectionSum / fBackgroundCollectionCyclesDone;
+END_IF
+// Getting wattage and energy
+IF fResponsivity <> 0 AND fFrequency <> 0 THEN
+    fResponWattage := (fVoltage - fBackground) / fResponsivity;
+    fResponMJ := fResponWattage / fFrequency;
+END_IF
```

Comment:
I think we should still calculate the Wattage even if the frequency is zero, because it's possible that our frequency information is actually wrong.

Maybe we should set the MJ to some obviously invalid value in these cases?